### ISSUE TITLE
clear cache with redis / group?

### ISSUE THREAD

**Issue by mster81** [arthurfranca/apicache-plus#28]
Hi, 

I'm having a bit of an issue with being able to clear the cache stored on redis. Looking at the docs I can see the recomended way to clear cache is to use apicacheGroup

`req.apicacheGroup = 'bookList'`

So, in my route, I have the apicachegroup at the beginning of the route. When I look at my redis instance, I see the group name that was created with the request. 

Now, when I try and 'clear' the cache using 

`apicache.clear('bookList')`

It appears to 'clear' the group that was created, however the actual data 'keys' remain, I also noticed that when the cache timeout hits, it will clear our the actual cached items, but leaves the group? 

I'm probably not setting this up correctly, however the docs make no mention of how to clear with redis so I'm a bit confused by this. 

Basically, the flow I'm looking to do is 

1) API returns JSON
2) User updates record on CMS
3) After update a purge to that specific cache 'group'
4) Next request will get updated data 

Thanks for any help

**Comment by arthurfranca** [comment]
@mster81 You are clearing it correctly. You may want to `await` it but it will block the response until clearing end, so it's optional.

The data keys stays there a little more, cause their values may be in use (in the middle of a response streaming) but their time to live gets decremented.
The actual group should be deleted too after each group's key. But it may be recreated right after by another request so maybe that's why you see it?

**Comment by mster81** [comment]
@arthurfranca Thanks for the info! 

I was actually able to get this working shortly after posting the question :D It was an issue on my end so I was bit premature in posting the question. 

What I ended up doing actually is right after clearing the group, I prime the page so as well, that way I can control the recaching a bit better, but its good to know how or why the keys still remained.
